scripts/src/scverse_template_scripts/cruft_prs.py

```diff
@@ -129,95 +133,208 @@ def matches_current_version(self, pr: PullRequest) -> bool:
+    upstream = clone.create_remote(name="upstream", url=original_repo.clone_url)
     upstream.fetch()
-    branch = clone.create_head(pr.branch, f"{pr.repo_id}/{origin.default_branch}")
-    branch.checkout()
 
-    run_cruft(path, tag_name=tag_name, log_name=pr.branch)
+    # Get the default branch
+    default_branch = original_repo.default_branch
+
+    # Get cruft config from the default branch in the upstream repo
+    log.info(f"Getting .cruft.json from the {default_branch} in {original_repo.clone_url}")
+    try:
+        # Try to get .cruft.json from the latest commit in upstream's default branch
+        cruft_content = clone.git.show(f"upstream/{default_branch}:.cruft.json")
+        cruft_config = json.loads(cruft_content)
+        log.info(f"Successfully read .cruft.json from upstream/{default_branch}")
+    except GitCommandError:
+        msg = "No .cruft.json found in repository"
+        raise FileNotFoundError(msg) from None
+
+    # Check if the branch already exists in the forked repo
+    remote_refs = [ref.name for ref in clone.remote().refs]
+    full_branch_name = f"refs/heads/{template_branch_name}"
+
+    # create and/or checkout template-update branch
+    if full_branch_name not in remote_refs:
+        log.info(f"Branch {template_branch_name} does not exists yet, creating it from initial commit")
+        # Get the initial commit on the default branch
+        initial_commit = next(clone.iter_commits(default_branch, reverse=True))
+
+        # Create and checkout a new branch from the initial commit
+        branch = clone.create_head(template_branch_name, initial_commit)
+        branch.checkout()
+    else:
+        log.info(f"Branch {template_branch_name} already exists, checking it out")
+        branch = clone.create_head(template_branch_name, f"origin/{template_branch_name}")
+        branch.checkout()
+
+    # Remove everything from the repo (except the `.git` directoroy)
+    cmd = ["/usr/bin/find", ".", "-not", "-path", "./.git*", "-delete"]
+    log.info("Running " + " ".join(cmd) + f" in {clone_dir}")
+    run(cmd, check=True, cwd=clone_dir)
+
+    # Initalize a new repo off the current template version, using the configuration from .cruft.json
+    template_dir = workdir / "template"
+    template_dir.mkdir()
+    cookiecutter_config = template_dir / "cookiecutter.json"
+    with cookiecutter_config.open("w") as f:
+        # need to put the cookiecutter-related info from .cruft.json into separate file
+        json.dump(cruft_config["context"]["cookiecutter"], f)
+    # run in a subprocess, otherwise not possible to catpure output of post-run hooks
+    with cruft_log_file.open("w") as log_f:
+        cmd = [
+            sys.executable,
+            "-m",
+            "cruft",
+            "create",
+            "https://github.com/scverse/cookiecutter-scverse",
+            f"--checkout={tag_name}",
+            "--no-input",
+            f"--extra-context-file={cookiecutter_config}",
```

Comment:
What happens if the `cookiecutter_config` doesn't have all variables that the template has? IIRC it just takes the defaults/options from the template cookiecutter.json and doesn't break, right? It's been a while...

Reply:
It takes the defaults from the cookiecutter.json.